### PR TITLE
Restructure of battery setup information

### PR DESCRIPTION
**Pull request by hamishwillee**
HI @MaEtUgR 

This is my modifications to your branch. I think it is an improvement and I hope I haven't lost any of your excellent information. 

I still need advice on:
1. The section "Calibration Voltage Divider & Current per volt" is a Basic Setting. Presumably though it is only relevant if you have a current measurement, and in which case you're probably using load compensation? 
   * What if you aren't using load compensation - is this just ignored?
   *Basically I want to work out why you need to do this and when.
1. BAT_CAPACITY - is a value in mA (according to docs). Is that correct - I would have though mAh (milliamp hours). Can you confirm value.

### CODE REVIEW

File: en/config/battery.md [hunk 1]
```diff
@@ -1,81 +1,151 @@
...
+> **Note** The battery monitoring features of PX4 can only be used if you have compatible hardware. In most cases this means a power module that measures the battery voltage, and may also measure the current between battery and vehicle.
 
-## Set the Basic Battery Settings
+## Overview
+
+The goal of the power setup is to provide a good estimate of remaining battery capacity, so that the vehicle is not used to the point that it runs out of power and crashes (or the battery is damaged due to deep-discharge).
+
+PX4 provides a number of (progressively more effective) methods that can be used to estimate the capacity:
+
+1. [Basic Battery Settings](#basic_settings) (default): raw measured voltage is compared to the range between "empty" and "full" voltages. This results in "course" estimates because measured voltage (and its corresponding capacity) will fluctuate under load.
```
Comment (ndepal) [inline en/config/battery.md]: should "course" be "coarse?

File: en/config/battery.md [hunk 2]
```diff
@@ -1,81 +1,151 @@
...
+1. [Voltage-based Estimation with Load Compensation](#load_compensation): Counteracts the effects of loading on the capacity calculation.
+1. [Voltage-based Estimation with Current Integration](#current_integration): Fuses the load-compensated voltage-based estimate for the available capacity with a current-based estimate of the charge that has been consumed. This results in a capacity estimate that is comparable to that of a smart battery.
+
+Later methods build on preceding methods. The approach you use will depend on whether the vehicle's power module can measure current.
+
+
+## Basic Battery Settings (default) {#basic_settings}
+
+> **Note** This default/basic power configuration results in relatively course estimations due to fluctuations in the estimated charge as the measured voltage changes under load.
```
Comment (ndepal) [inline en/config/battery.md]: ditto

File: en/config/battery.md [hunk 3]
```diff
@@ -1,81 +1,151 @@
...
+If you have a vehicle that measures voltage through a power module and the ADC of the flight controller then you should check and calibrate the measurements once per board. To calibrate you'll need a multimeter.
+
+The easiest way to calibrate the divider is by using *QGroundControl* and following the step-by-step guide on [Setup > Power Setup](https://docs.qgroundcontrol.com/en/SetupView/Power.html) (QGroundControl User Guide).
+
+
+### Amps per volt {#current_divider}
+
+> **Tip** This setting is not needed if you are using the basic configuration (without load compensation etc.)
+
+If you are using [Current-based Load Compensation](#current_based_load_compensation) or [Current Integration](#current_integration) the APMs per volt needs to be calibrated.
```
Comment (ndepal) [inline en/config/battery.md]: Typo APMs -> amps

File: en/config/battery.md [hunk 4]
```diff
@@ -1,81 +1,151 @@
...
+The easiest way to calibrate the dividers is by using *QGroundControl* and following the step-by-step guide on [Setup > Power Setup](https://docs.qgroundcontrol.com/en/SetupView/Power.html) (QGroundControl User Guide).
+
+
+## Voltage-based Estimation with Load Compensation {#load_compensation}
+
+> **Note** With well configured load compensation the voltage used for battery capacity estimation is much more stable, varying far less when flying up and down.
+
+Load compensation attempts to counteract the fluctuation in measured voltage/estimated capacity under load that occur when using the [basic configuration](#basic_settings). This works by estimating what the voltage would be for the *unloaded* battery, and using that voltage (instead of the measured voltage) for estimating the remaining capacity. 
+
+> **Note** To use the load compensation you will still need to set the [basic configuration](#basic_settings). The *Empty Voltage* ([BAT_V_EMPTY](../advanced_config/parameter_reference.md#BAT_V_EMPTY)) should be set higher than because the compensated voltage gets used for the estimation (typically set a bit below the expected rest cell voltage when empty after use).
```
Comment (ndepal) [inline en/config/battery.md]: missing word? "should be set higher than because"

File: en/config/battery.md [hunk 5]
```diff
@@ -1,81 +1,151 @@
...
+
+### Current-based Load Compensation (recommended) {#current_based_load_compensation}
+
+This load compensation method relies on current measurement to determine load. It is far more accurate than [Thrust-based Load Compensation](#thrust_based_load_compensation) but requires that you have a current sensor.
+
+To enable this feature:
+
+1. Set the parameter [BAT_R_INTERNAL](../advanced_config/parameter_reference.md#BAT_R_INTERNAL) to to the internal resistance of your battery. 
+   > **Tip** There are LiPo chargers out there which can measure the internal resistance of your battery. A typical value is 5mΩ but this can vary with discharge current rating, age and health of the cells.
+1. You should also calibrate the [Amps per volt divider](#current_divider) in the basic settings screen.
```
Comment (ndepal) [inline en/config/battery.md]: What parameter does this affect within PX4?

**Comment by MaEtUgR** [comment]
1.
> Any measurement you want to use needs to be calibrated:
> - For basic voltage based estimation the voltage divider needs to be calibrated.
> - If you use the current for load compensation or accumulation of capacity via current the current per volt needs to be calibrated.
2. You're right, that was an error. https://github.com/PX4/Firmware/pull/8950

File: en/config/battery.md (hunk 1 above)
Comment (hamishwillee) [reply en/config/battery.md]: Yes. Thanks. Will fix in both places.

File: en/config/battery.md (hunk 3 above)
Comment (hamishwillee) [reply en/config/battery.md]: Thanks!

File: en/config/battery.md (hunk 4 above)
Comment (hamishwillee) [reply en/config/battery.md]: Yes "than without compensation". Will fix thanks.

File: en/config/battery.md (hunk 5 above)
Comment (hamishwillee) [reply en/config/battery.md]: @MaEtUgR That is a good question - ?

**Comment by hamishwillee** [comment]
@ndepal Thanks for the review. I have fixed those typos. Hope you find this clear.

@MaEtUgR Comment above on the calibration/dividers has been integrated.

If you are OK with this, please merge the PR into your branch - because we really should make the "main PR" on the px4 repo the focus for discussion rather than here. This can be done before we know about the parameters affected by the dividers.

File: en/config/battery.md [hunk 6]
```diff
@@ -1,81 +1,141 @@
...
+* [BAT_V_LOAD_DROP](../advanced_config/parameter_reference.md#BAT_V_LOAD_DROP) - [Thrust-based Load Compensation](#thrust_based_load_compensation).
+
+### Current-based Load Compensation (recommended) {#current_based_load_compensation}
+
+This load compensation method relies on current measurement to determine load. It is far more accurate than [Thrust-based Load Compensation](#thrust_based_load_compensation) but requires that you have a current sensor.
+
+To enable this feature:
+
+1. Set the parameter [BAT_R_INTERNAL](../advanced_config/parameter_reference.md#BAT_R_INTERNAL) to to the internal resistance of your battery. 
+   > **Tip** There are LiPo chargers out there which can measure the internal resistance of your battery. A typical value is 5mΩ but this can vary with discharge current rating, age and health of the cells.
```
Comment (MaEtUgR) [inline en/config/battery.md]: We should write 5mOhm per cell... It's somehow implicitly readable but I also never wrote it explicit.

File: en/config/battery.md [hunk 7]
```diff
@@ -1,81 +1,141 @@
...
-Best suitable measurement to infer charge status of the cell.
-* Load compensated voltage calculated in air (under load)
-Ideally you want to have a load compensated voltage available during flight which reflects the rest voltage without load. but in practise the load compensated value will always be a bit below what you end up with because the battery "recovers" its unloaded voltage over time without gaining usable capacity.
+1. Set the parameter [BAT_V_LOAD_DROP](../advanced_config/parameter_reference.md#BAT_V_LOAD_DROP) to how much voltage drop a cell shows under the load of full throttle.
 
-## Without load compensation
-With the default parameter set there is no load compensation enabled because if you don't have a current sensor, it might be uncalibrated or the internal resistance of your battery is completely off the resulting compensation might not make sense.
 
-If you don't want to use load compensation set the empty cell voltage to the value that should result in 0% battery estimation.
+## Voltage-based Estimation with Current Integration {#current_integration}
```
Comment (MaEtUgR) [inline en/config/battery.md]: Is this title clear enough? Probably I need to think of a better title...
Maybe something like "Voltage-based Estimation fused with Current-based capacity accumulation"?
I'm not sure about that, if it's understandable we can also let it like it is.

File: en/config/battery.md [hunk 8]
```diff
@@ -1,81 +1,141 @@
...
+The estimate of the charge that has been consumed over time is produced by mathematically integrating the measured current (this approach provides very accurate energy consumption estimates).
 
-## Combning voltage based estimation with current integration
-After tuning the voltage based estimation you can achieve a more stable linear and accurate estimation by fusing your estimate with the so far used capacity. This is only possible with accurate current measurements because the current gets mathematically integrated over time to have a used capacity estimate.
-This is the best way to have totally accurate relative fuel consumption of the battery. You can not use only this technique because:
-* You cannot detect how full the battery is when the vehicle is turned on
-* If the absolute fuel estimation is for some reason wrong or the battery is faulty you don't have feedback.
+At system startup PX4 first does uses a voltage-based estimate to determine the initial battery charge. This estimate is then fused with the value from current integration to provide a combined better estimate.
```
Comment (MaEtUgR) [inline en/config/battery.md]: typo? "does uses"

File: en/config/battery.md [hunk 9]
```diff
@@ -1,81 +1,141 @@
...
-> **Warning** Before you fly for the first time you should make sure you have a way to monitor your battery state. Not knowing how much fuel is left may likely lead to deep discharge of the used battery which can cause damage to the battery, loss of vehicle control and hence a crash.
+This topic explains how to configure power settings.
 
-> **Note** The battery monitoring related features of PX4 can only be used if you have compatible hardware which is in most cases a power module that measures the battery voltage and the current between battery and vehicle.
+> **Note** The battery monitoring features of PX4 can only be used if you have compatible hardware. In most cases this means a power module that measures the battery voltage, and may also measure the current between battery and vehicle.
 
-## Set the Basic Battery Settings
+## Overview
+
+The goal of the power setup is to provide a good estimate of remaining battery capacity, so that the vehicle is not used to the point that it runs out of power and crashes (or the battery is damaged due to deep-discharge).
```
Comment (MaEtUgR) [inline en/config/battery.md]: This is only nit-pick I think there's noting wrong but when suing the word **capacity** there's no clear differentiation between:
- Battery percentage, charge, fuel, ratio of capacity left, no unit
- Remaining capacity in mAh

The first is what PX4 is aiming to provide to the pilot as accurate as possible because it's the indicator in the UI and the second is only available in the last method with current integration and would allow for flight time estimate.

File: en/config/battery.md (hunk 5 above)
Comment (MaEtUgR) [reply en/config/battery.md]: `BAT_A_PER_V`

File: en/config/battery.md (hunk 8 above)
Comment (MaEtUgR) [reply en/config/battery.md]: I already fixed this one after merging https://github.com/PX4/px4_user_guide/pull/131/commits/b3e6e555c02c1fb0d40a4168e8b1ba247390103b

File: en/config/battery.md (hunk 7 above)
Comment (hamishwillee) [reply en/config/battery.md]: I have made the very minor modification to add "Fused" in the title (in another PR):

> Voltage-based Estimation Fused with Current Integration

Your proposal is more accurate, but it is visually distracting and harder to parse really long titles. I'd rather make sure that people get the "gist" of the topic quickly, and explain in more detail in the body. The reason I chose to add "fused" is that this usefully captures the concept that the two methods are, well, fused :-)

File: en/config/battery.md (hunk 8 above)
Comment (hamishwillee) [reply en/config/battery.md]: Thanks

File: en/config/battery.md (hunk 9 above)
Comment (hamishwillee) [reply en/config/battery.md]: You're quite right - I was deliberately not making the distinction because I don't think it is important at this point (the overview). That said, they it doesn't hurt to make the point that there "is a distinction". So I have changed this to "remaining battery percentage (and capacity)".
